test/jdk/java/util/regex/RegExTest.java

```diff
@@ -223,43 +104,37 @@ private static String getRandomAlphaString(int length) {
 
     private static void check(Matcher m, String expected) {
         m.find();
-        if (!m.group().equals(expected))
-            failCount++;
+        assertEquals(m.group(), expected);
     }
 
     private static void check(Matcher m, String result, boolean expected) {
         m.find();
-        if (m.group().equals(result) != expected)
-            failCount++;
+        assertEquals(m.group().equals(result), expected);
     }
 
     private static void check(Pattern p, String s, boolean expected) {
-        if (p.matcher(s).find() != expected)
-            failCount++;
+        assertEquals(p.matcher(s).find(), expected);
     }
 
     private static void check(String p, String s, boolean expected) {
         Matcher matcher = Pattern.compile(p).matcher(s);
-        if (matcher.find() != expected)
-            failCount++;
+        assertSame(matcher.find(), expected);
```

Comment:
Why use `assertSame(Object, Object)`? I would expect `assertEquals(boolean, boolean)`.

Reply:
Artifacting because of the use of `==` I'll make it more readable.